src/internal/snapshot/server.go

```diff
@@ -0,0 +1,18 @@
+package snapshot
+
+import (
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/fileset"
+	snapshotserver "github.com/pachyderm/pachyderm/v2/src/snapshot"
+)
+
+type Env struct {
+	DB    *pachsql.DB
+	store *fileset.Storage
+}
+
+func NewAPIServer(env Env) snapshotserver.APIServer {
```

Comment:
Overall, I see these constructors as being pretty low-value.  If computation needs to be done every time you create an instance, that's fine, but if you're just putting positional arguments into a struct, they only make the code harder to read.